internal/workloadstatus/errortypes.go

```diff
@@ -0,0 +1,89 @@
+package workloadstatus
```

Comment:
do we really need a custom error type for every situation?
ContainerNotRunningError and PodIsPendingError and PodIsFailedError are the same thing,
a single type with an additional Type Field could make it easier to maintain.